mat/transform_test.go

```diff
@@ -0,0 +1,163 @@
+package mat
+
+import (
+	"math"
+	"testing"
+)
+
+func TestTranslate(t *testing.T) {
+	m := Translate(1, 2, 3)
+	v := Vec3{4, 5, 6}
+	out := m.Transform(v)
+	expected := Vec3{5, 7, 9}
+
+	if !out.Equal(expected) {
+		t.Errorf("%s transformed by %s is expected to be %s, got %s",
+			v, m, expected, out,
+		)
+	}
+}
+
+func TestRotate(t *testing.T) {
```

Comment:
I'm not sure if this is really important but these tests do not explicitly validate the content of the rotation matrix. The only check the result after applying the rotation to a vector. As `Transform` already has some tests in https://github.com/seqsense/pcgol/blob/ca56a24aab36ff72a5b5cc0260b4192a8bbb7575/mat/mat4_test.go#L133, it is probably not a big deal.

Reply:
Test writing down the expected matrix element isn't human readable and goes like a snapshot test. Snapshot test can't detect mistake like transposed rotation matrix.